File: frontend/src/types/entities/cliente.ts
export interface Persona {
  nombre: string;
  documento: string;
  direccion: string | null;
  cp: string | null;
  poblacion: string | null;
  provincia: string | null;
  pais: string | null;
  email: string | null;
  telefono: string | null;
}

export interface ClientePayload extends Persona {
  titular: string | null;
}

export interface ClienteEditData extends ClientePayload {
  id?: number;
}

export interface Cliente extends ClientePayload {
  id?: number;
}

File: src/main/java/com/jcfp/tallererp/entity/Cliente.java
package com.jcfp.tallererp.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity(name = "clientes")
@Getter
@Setter
public class Cliente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String nombre;
    private String titular;
    @Column(unique = true, nullable = false)
    private String documento;
    private String direccion;
    private String cp;
    private String poblacion;
    private String provincia;
    private String pais;
    private String email;
    private String telefono;
}
